static/specification.html

```diff
@@ -138,605 +336,555 @@ <h2>
+              <var>element</var>'s <code data-x=
+              "attr-link-crossorigin">crossorigin</code> content attribute.
+            </p>
+          </li>
+          <li>
+            <p>
+              Return true.
+            </p>
+          </li>
+        </ol>
+        <p>
+          To <span data-x="process the linked resource">process this type of
+          linked resource</span> given a <code>link</code> element
+          <var>element</var>, boolean <var>success</var>, and <span data-x=
+          "concept-response">response</span> <var>response</var>:
+        </p>
+        <ol>
+          <li>
+            <p>
+              If |response|'s status is not an OK status, the set |success| to
+              false.
+            </p>
+          </li>
+          <li>
+            <p>
+              Otherwise, if <var>response</var>'s <span data-x=
+              "Content-Type">Content-Type metadata</span> is not a
+              <code>application/json</code>, then set <var>success</var> to
+              false.
+            </p>
+          </li>
+          <li>Let |json| be the result of [=parse JSON bytes to an Infra
+          value=] passing |response|'s [=Response/body=].
+          </li>
+          <li>If |json| is an exception, then set <var>success</var> to false.
+          </li>
+          <li>
+            <p>
+              If the user agent has exhausted the number of allowed [=payment
+              sessions=], set |success| to false.
+            </p>
+          </li>
+          <li>Otherwise, establish a new [=payment session=].
+          </li>
+          <li>[=Queue an element task=] on the [=networking task source=] given
+          |element| and the following steps:
+            <ol>
+              <li>If |success| is true, [=fire an event=] named `"load"` at
+              |element|.
+              </li>
+              <li>Otherwise, [=fire an event=] named `"error"` at |element|.
+              </li>
+            </ol>
+          </li>
+        </ol>
+      </div>
+      <p>
+        If a "monetization" <code>link</code> <span>becomes browsing-context
+        disconnected</span>, a user agent MUST stop the [=payment session=].
+      </p>
     </section>
-    <section id="flow">
+    <section>
       <h2>
-        Flow
+        `onmonetization` event handler
       </h2>
       <p>
-        This flow refers to the <a>user agent</a> and the user's
-        <a>provider</a>.
+        The `onmonetization` event handler is an [=event handler content
+        attribute=] that can be applied to any [=element=]. The user agent uses
+        it to notify that some [^link^] has been [=monetized=].
       </p>
-      <ol>
-        <li>The <a>user agent</a> sets {{ Monetization / state }} to "{{
-        MonetizationState / pending }}".
+    </section>
+    <section id="events">
+      <h2>
+        monetization event
+      </h2>
+      <p>
+        When the [=payment session=] has sent an |outgoing payment| with a
```

Comment:
This means only one event will be fired, not upon each packet ? 
Or one tag configuration could fire multiple outgoing payments? 
Switch tabs and it pauses. 
Return to the tab 30 minutes later and it perhaps uses a new outgoing payment? 
It's a bit unclear to me where the payment is happening, from an OP server, or ILP running directly in the browser. 

Links to relevant OP specs for referenced concepts might be good (is there one in this doc already) ?

Reply:
@adrianhopebailie
A monetization event per outgoing_payment rather than per packet ? 

Probably a question for you

Reply:
This should be per **payment**, whatever that means. I believe that it should depend on the WM Provider.

i.e. If the provider sends 5 small packets and then emits 1 event to the browser then the website only gets one event. If the provider emits 5 events then the browser should emit 5 events IMHO.

Reply:
> It's a bit unclear to me where the payment is happening, from an OP server, or ILP running directly in the browser.

In @AlexLakatos  model, as far as I can tell, the browser creates the STREAM itself, having been given a connection url/rate/token from the Provider component. (It's unclear to me how it can usefully give a rate based on just connection details to keep with the privacy model).

@adrianhopebailie 
It seems to me, in your model, the WM Provider could create a STREAM itself, in which case, it would likely emit an event for each packet, but it could also be sending payments via http requests, in which case there would just be one packet unless it polls, in order to support progress events. 

A website showing a counter (debatable how much value that really has, but let's just say a website frontend wants the events for some reason) would have to poll the incomingPayment url for updates, if it can't somewhat reliably assume that it will get progress events. 

Could OP support subscriptions (server push, websockets etc) at an incoming payment ?